src/haz3lcore/statics/ConstructorMap.re

```diff
@@ -194,7 +202,8 @@ let map = (f: option('a) => option('b), m: t('a)): t('b) => {
 let get_entry = (ctr, m) =>
   List.find_map(
     fun
-    | Variant(ctr', _, value) when Constructor.equal(ctr, ctr') => value
+    | Variant(ctr', _, value) when Constructor.equal(ctr, ctr') =>
```

Comment:
we might want to make an issue to turn at least one of these options into a custom type... strictly speaking this is an improvement but a double option is likely to cause confusion